### PR TITLE
Update flatpak manifest json and shared-modules submodule

### PR DESCRIPTION
**Pull request by rcalixte**
Add libnotify as notifications do not consistently work without it and update shared-modules submodule

### CODE REVIEW

File: io.github.Pithos.json [hunk 1]
```diff
@@ -19,7 +19,14 @@
...
 
-    "--talk-name=org.kde.StatusNotifierWatcher"
+    "--talk-name=org.kde.StatusNotifierWatcher",
+    "--talk-name=org.freedesktop.Notifications",
+
+    "--talk-name=org.gnome.SessionManager",
+    "--talk-name=org.xfce.SessionManager",
+
+    "--talk-name=org.mpris.MediaPlayer2.io.github.Pithos",
+    "--own-name=org.mpris.MediaPlayer2.io.github.Pithos"
```
Comment (TingPing) [inline io.github.Pithos.json:29]: These are default permissions and can be removed.

File: io.github.Pithos.json [hunk 2]
```diff
@@ -19,7 +19,14 @@
     "--talk-name=org.gnome.SettingsDaemon.MediaKeys",
     "--talk-name=org.mate.SettingsDaemon",
 
-    "--talk-name=org.kde.StatusNotifierWatcher"
+    "--talk-name=org.kde.StatusNotifierWatcher",
+    "--talk-name=org.freedesktop.Notifications",
```
Comment (TingPing) [inline io.github.Pithos.json:23]: This is incorrect.

File: io.github.Pithos.json [hunk 3]
```diff
@@ -34,15 +41,34 @@
         "sha256": "e6e3de4e1f3b201814a956ab8f16dfc8a262db1937ff1eee4d855365398c6020"
       }]
     },
+    {
```
Comment (TingPing) [inline io.github.Pithos.json:44]: This isn't used.

File: io.github.Pithos.json [hunk 4]
```diff
@@ -34,15 +41,34 @@
...
     {
       "name": "pithos",
       "buildsystem": "meson",
       "cleanup": ["/share/man"],
       "sources": [
         {
           "type": "git",
-          "url": "https://github.com/pithos/pithos.git",
-          "commit": "1.5.1"
+          "url": "https://github.com/rcalixte/pithos.git",
```
Comment (TingPing) [inline io.github.Pithos.json:70]: What?

Comment (rcalixte) [reply io.github.Pithos.json:70]: testing the build... this was a temporary commit

**Comment by rcalixte** [comment]
@TingPing I'm trying to address an issue where the flatpak version of Pithos does not seem to work with system notifications. Adding `--talk-name=org.freedesktop.Notifications` to the `flatpak run` command yields an initial notification but no consecutive ones. Building from source on the same system has working notifications. I can share more details but that is the current issue I am trying to debug.

**Comment by TingPing** [comment]
Pithos does not use that interface, it uses the notification portal.

**Comment by TingPing** [comment]
You may have issues in the xdg-desktop-portal-gtk service.

**Comment by rcalixte** [comment]
@TingPing I saw the changes in the code to support that but there's no denying that the changing flag has an effect. I've looked into my desktop environment to see if it is may forwarding things but I'm still early in debugging this deeply. I did check everything yesterday with regards to xdg-desktop-portal* and I wasn't able to find anything there yet.

**Comment by TingPing** [comment]
Even if you get Pithos to use that interface it is not a change I want here. The portal is the method we should use and bugs there should be resolved there.

**Comment by rcalixte** [comment]
Can you think of any other reasons why the flatpak version of the app would have notification issues? I have tried other media apps via flatpak for troubleshooting purposes and those had no issues with notifications either. Would it be a bad thing to expand compatibility for this package?

**Comment by rcalixte** [comment]
@TingPing I'm likely going to wait until the weekend to debug this more deeply but I'm open to whatever routes of troubleshooting you might suggest beyond what I've already done. If it's easier to track this via an issue, I can do that.

**Comment by TingPing** [comment]
Almost no project other than Pithos uses the portal  so thats what makes it unique and again likely where the issue lies.

**Comment by rcalixte** [comment]
But it's not the application that's the issue? Notifications work fine when it is compiled from source on the same machines. So the issue is isolated with the flatpak. Is there a chance that there is a missing flatpak dependency?

**Comment by rcalixte** [comment]
@TingPing But it's not the application that's the issue? Notifications work fine when it is compiled from source on the same machines. So the issue is isolated with the flatpak. Is there a chance that there is a missing flatpak dependency?

**Comment by TingPing** [comment]
The portal is only used in flatpak. I understand  your frustration but I don't like the solution of bypassing the portal.

**Comment by rcalixte** [comment]
@TingPing On which distributions have you tested the flatpak version? I tested from the default in Ubuntu 18.04 and 20.04 (not my daily driver DE/WM) and notifications didn't work properly. I've tested with the flatpak versions available in the default repositories and in the PPA.

**Comment by TingPing** [comment]
Just testing it I can reproduce that it doesn't appear to function. So there is a real bug somewhere.

**Comment by rcalixte** [comment]
@TingPing I'll take a look and see what I can find on my end too. The new notification API is fairly new to me so I need to get up to speed. I'll also test any patches if you have a build you're confident about. Thanks for confirming though!

**Pull request closed without merging**